runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp

```diff
@@ -653,10 +653,156 @@ TR::Register *TR::ARM64PrivateLinkage::buildDirectDispatch(TR::Node *callNode)
    return retReg;
    }
 
+static TR::Register *evaluateUpToVftChild(TR::Node *callNode, TR::CodeGenerator *cg)
+   {
+   TR::Register *vftReg;
+   uint32_t firstArgumentChild = callNode->getFirstArgumentIndex();
+   for (uint32_t i = 0; i < firstArgumentChild; i++)
+      {
+      TR::Node *child = callNode->getChild(i);
+      vftReg = cg->evaluate(child);
+      cg->decReferenceCount(child);
+      }
+   return vftReg;
+   }
+
+void TR::ARM64PrivateLinkage::buildVirtualDispatch(TR::Node *callNode,
+   TR::RegisterDependencyConditions *dependencies,
+   uint32_t argSize)
+   {
+   TR::CodeGenerator *codeGen = cg();
+
+   TR::Register *x0 = dependencies->searchPreConditionRegister(TR::RealRegister::x0);
+   TR::Register *x9 = dependencies->searchPreConditionRegister(TR::RealRegister::x9);
+
+   TR::SymbolReference *methodSymRef = callNode->getSymbolReference();
+   TR::MethodSymbol *methodSymbol = methodSymRef->getSymbol()->castToMethodSymbol();
+   TR::LabelSymbol *doneLabel = NULL;
+
+   TR::Instruction *gcPoint;
+
+   TR_J9VMBase *fej9 = (TR_J9VMBase *)(comp()->fe());
+
+   // Computed calls
+   //
+   if (methodSymbol->isComputed())
+      {
+      // ToDo: Implement this path (Eclipse OpenJ9 Issue #7023)
+      comp()->failCompilation<TR::AssertionFailure>("ComputedCall");
+      }
+
+   // Virtual and interface calls
+   //
+   TR_ASSERT(methodSymbol->isVirtual() || methodSymbol->isInterface(), "Unexpected method type");
+
+   void *thunk = fej9->getJ2IThunk(methodSymbol->getMethod(), comp());
+   if (!thunk)
+      thunk = fej9->setJ2IThunk(methodSymbol->getMethod(), TR::ARM64CallSnippet::generateVIThunk(callNode, argSize, codeGen), comp());
+
+   if (methodSymbol->isVirtual())
+      {
+      TR::MemoryReference *tempMR;
+      if (methodSymRef->isUnresolved() || comp()->compileRelocatableCode())
+         {
+         TR::Register *vftReg = evaluateUpToVftChild(callNode, codeGen);
```

Comment:
These two lines could move outside the `if` because the same code is used on the `else` part as well.

Reply:
Moved and commoned those lines out of `if`.